Self Hosted demo app

colombod opened the pull request:
the PiTop.Interactive.Rover app is hostign the rover loop and embeds .NET Interactive

the kernel is exposed via signalR

jonsequitur commented on src/Examples/PiTop.Interactive.Rover/CommandLine/CommandLineParser.cs:
```diff
@@ -0,0 +1,206 @@
+            StartHttp startHttp = null,
+            Action onServerStarted = null)
+        {
+            var operation = Log.OnEnterAndExit();
```
```suggestion
            using var operation = Log.OnEnterAndExit();
```

jonsequitur commented on src/Examples/PiTop.Interactive.Rover/RoverRobotSetup.cs:
```diff
@@ -0,0 +1,116 @@
+            });
+        }
+
+        private static async Task RegisterTypes<T>(CSharpKernel csharpKernel)
```
This method name is a misleading.

jonsequitur commented on src/Examples/PiTop.Interactive.Rover/RoverRobotSetup.cs:
```diff
@@ -0,0 +1,116 @@
+            await RegisterTypes<FoundationPlate>(csharpKernel);
+            await RegisterTypes<ExpansionPlate>(csharpKernel);
+            await RegisterTypes<RoverRobot>(csharpKernel);
+            await csharpKernel.SetVariableAsync("roverRobot", rover);
```
I would align these names and use `nameof(rover)` to make things a little clearer.

jonsequitur commented on src/Examples/PiTop.Interactive.Rover/RoverRobotSetup.cs:
```diff
@@ -0,0 +1,116 @@
+            {
+                source.Cancel(false);
+                Task.WaitAll(new []{robotLoop, reactLoop},TimeSpan.FromSeconds(10));
+                if (csharpKernel.TryGetVariable("roverRobot", out RoverRobot roverRobot))
```
You have a reference to the instance above, so you shouldn't need to pull it out this way to dispose it.

Likely it could be registered up front for disposal as well.

colombod committed:
package upgrade


prototyping a self hosting asp.net


update packages


inject robot in c# kernel


inject rover robot


with loop


samples


addressing pr feedback


implement stdio server for backward compatibility


ml module


filter on threshold


add resizing utility


tooling for example and app for demo


improvements


updates


no camera stdout


refactor setup code


revert update


remove package
